src/test/run-pass/single-derive-attr-allow-unstable.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// ignore-pretty : (#23623) problems when  ending with // comments
+
+#![feature(allow_internal_unstable)]
+
+#[allow_internal_unstable]
+macro_rules! expand_to_unstable {
+    () => {
+        // FIXME(eddyb) #[allow_internal_unstable]
+        // doesn't actually work for some reason.
+
+        #[derive(Clone)] //#[derive_Clone]
```

Comment:
There is, `custom_derive`, but `allow_internal_unstable` is supposed to bypass the actual feature gating (i.e. any feature can be used inside an `#[allow_internal_unstable]` macro).
This works with, e.g. `#[thread_local] static FOO: usize = 0;`, but not `#[derive_Clone]`.